File: problem/7/python/solve_7.py
#!/usr/bin/env python
# http://projecteuler.net/problem=7

candidates = range(2,150000)

primes = []

while candidates:
        primes.append(candidates.pop(0))
        print primes[-1]
        candidates = filter(lambda x: x % primes[-1], candidates)

print primes[-1]
print primes[10000]

File: problem/7/perl/solve_7.pl
#!/usr/bin/env perl6
# http://projecteuler.net/problem=7

use v6;

my @candidates = (2 .. 149999);

my @primes;

#say Array.^methods>>.name.perl;
# (@candidates Z=> @candidates.keys).hash.<$c>
loop (;@candidates.elems > 0;) {
        @primes.push(@candidates.shift());
        say @primes[* - 1];
#        for 0 .. @candidates.end() -> $c {
#                if @candidates[$c] % @primes[* - 1] == 0 {
#                        @candidates.delete($c);
#                }
#        }
        @candidates = grep { $_ % @primes[* - 1] != 0 }, @candidates;
}

say "elements: " ~ @primes.elems;
say "last prime: " ~ @primes[* - 1];
say "10001: " ~ @primes[10000];
